app/static/js/createEvents.js

```diff
@@ -60,18 +60,86 @@ function format24to12HourTime(timeStr){
       });
 
   }
+
+
+function getSelectedFiles(){
+  let _fileHolder = new DataTransfer();
+  $(".fileHolder").each(function(){
+    _fileHolder.items.add($(this).data("file"));
+  });
+  return _fileHolder.files;
+}
+
+
+function hasUniqueFileName(fileName){
+  let nameHolders = $(".fileName").toArray();
+  for (let i = 0; i < nameHolders.length; i++){
+    if (fileName == $(nameHolders[i]).data("filename")) return false;
+  }
+  return true;
+}
+
+
+
 /*
  * Run when the webpage is ready for javascript
  */
 $(document).ready(function() {
   if ( $("#startDatePicker").val() != $("#endDatePicker").val()){
-
     calculateRecurringEventFrequency();
   }
 
-    $("#attachmentObject").fileinput({
-        allowedFileExtensions:["pdf","jpg","png","gif", "csv", "docx", "jpg", "jpeg", "jfif"]
-    })
+    var fileNum = 0;
+    $("#attachmentObject").on('change', function() {
+      const selectedFiles = $("#attachmentObject").prop('files');
+      for (let i = 0; i < selectedFiles.length; i++) {
+        const file = selectedFiles[i];
+        if (hasUniqueFileName(file.name)){
+          let fileName = (file.name.length > 25) ? file.name.slice(0,10) + '...' + file.name.slice(-10) : file.name;
+          let fileExtension = file.name.split(".").pop();
+          let iconClass = '';
+          switch(fileExtension) {
+            case 'jpg': 
+            case 'png':
+            case 'jpeg':
+              iconClass = "bi-file-image";
+              break
+            case 'pdf':
+              iconClass = 'bi-filetype-pdf';
+              break
+            case 'docx':
+              iconClass = 'bi-filetype-docx';
+              break
+            case 'xlsx':
+              iconClass = 'bi-filetype-xlsx';
+              break
+            default:
+              iconClass = 'bi-file-earmark-arrow-up';
+          }
+          $("#attachedObjectContainer").append("<div class='border row p-0 m-0' id='attachedFilesRow" +fileNum+"'></div>")
+          $("#attachedFilesRow"+fileNum).append("<i class='col-auto fs-3 px-3 bi " + iconClass + "'></i>")
+          $("#attachedFilesRow"+fileNum).append("<div id='attachedFile" + fileNum + "' data-filename='" + file.name + "' class='fileName col-auto pt-2'>" + fileName + "</div>");
+          $("#attachedFilesRow"+fileNum).append("<div class='col' style='text-align:right'> \
+                                                  <div class='btn btn-danger fileHolder p-1 my-1 mx-1' id='trash" + fileNum + "'>\
+                                                    <span class='bi bi-trash fs-6'></span>\
+                                                  </div>\
+                                                </div>")
+          $("#trash"+fileNum).data("file", file);
+          $("#trash"+fileNum).on("click", function() {
+            let elementFileNum = $(this).prop("id").split("trash")[1];
```

Comment:
You don't need to get the file num out of the id here, because you already know it. You've just referenced it - `fileNum`.

Reply:
Oh! Actually, that variable is used to get the value from within a callback function so we don't have the luxury of using fileNum here. We have refactored that line though to make it more readable via an extra data attribute.